sentinel.go

```diff
@@ -162,31 +162,22 @@ func (s *Sentinel) defaultPool(addr string) *redis.Pool {
 	}
 }
 
-func (s *Sentinel) get(addr string) redis.Conn {
-	pool := s.poolForAddr(addr)
-	return pool.Get()
-}
-
 func (s *Sentinel) poolForAddr(addr string) *redis.Pool {
 	s.mu.Lock()
-	if s.pools == nil {
-		s.pools = make(map[string]*redis.Pool)
-	}
-	pool, ok := s.pools[addr]
-	if ok {
-		s.mu.Unlock()
+	defer s.mu.Unlock()
+	if pool, ok := s.pools[addr]; ok {
 		return pool
 	}
 	s.mu.Unlock()
```

Comment:
Since there is a `defer` we don't need this Unlock anymore?

I am not sure, but I think the initial idea of lock usage in this func could be avoid locking while calling  `s.newPool(addr)` (which can contain user supplied function to get Pool). This pr changes behaviour a bit.

Reply:
The original behaviour is still there, isn't it?

```
s.mu.Unlock()
newPool := s.newPool(addr)
s.mu.Lock()
```

Reply:
Oh, I see, was confused by defer and thought that the entire function meant to be under lock now, never seen this pattern before - interesting approach